--- server/net/minecraft/core/item/Item.java	2025-11-30 15:27:06
+++ inject/Item.java	2025-11-30 15:48:33
@@ -63,7 +63,7 @@
   protected final Map<String, Stat> stats = new HashMap<>();
   
   public Item(NamespaceID namespaceId, int id) {
-    this.maxStackSize = 64;
+    this.maxStackSize = 100;
     this.maxDamage = 0;
     this.hasSubtypes = false;
     this.containerItem = null;
@@ -307,4 +307,4 @@
   public static Item getItem(int id) {
     return itemsList[Math.max(id, 0)];
   }
-}
+}
\ No newline at end of file
